Lesson1 (Sequence containers)/library.cpp

```diff
@@ -0,0 +1,50 @@
+#include <iostream>
+#include "library.h"
+
+
+
+std::vector<int> combineVectors (const std::vector<int>::iterator& begin1, const std::vector<int>::iterator& end1, const std::vector<int>::iterator& begin2, const std::vector<int>::iterator& end2)
+{
+    ulong size = (end1 - begin1) + (end2 - begin2);
+    std::vector<int> tmp;
+    tmp.reserve(size);
+    tmp.insert(tmp.begin(), begin1, end1);
+    tmp.insert(tmp.end(),begin2, end2);
+
+    return tmp;
+};
+
+size_t magicNumber(const size_t number, const size_t degree)
+{
+    long wantedNumber = 0;
+    std::list<long> numbers;
+    numbers.push_back(number);
+
+    for (auto&& it = numbers.begin(); it != numbers.end(); ++it) {
+        if (numbers.size() < degree)
+            numbers.push_back(*(--numbers.end()) * *(numbers.begin()));
+
+        long first = *it;
+        long last = (*it) % 10;
+
+        while (first > 9)
```

Comment:
_please use braces always, even if it is one line expression
if (expr) {
// one line
}_